nuxeo-page-provider-display-behavior.html

```diff
@@ -514,15 +514,37 @@
       _fetchRange(firstIndex, lastIndex, clear) {
         if (this._hasPageProvider()) {
 
+          if (firstIndex === 0) {
+            lastIndex = this._pageSize - 1;
+          }
+
           if (!firstIndex && !lastIndex && !clear) {
             firstIndex = 0;
-            lastIndex = this._pageSize;
+            lastIndex = this._pageSize - 1;
             clear = true;
           }
 
           if (this.maxItems && lastIndex > this.maxItems) {
             lastIndex = this.maxItems;
             clear = true;
+          } else if (firstIndex > 0) {
+            lastIndex += 10;
```

Comment:
Same question as above, about overriding the value of `lastIndex`.

Reply:
Here, sometimes the visible range excludes items that are half visible (reproducible with grid). The user experience is bad because you start seeing items that are not loaded and will be loaded once fully visible (entirely on the screen). Adding this 10 value makes the scrolling more fluent because it preloads items that you are potentially about to see by keeping scrolling.

Reply:
I think we should leave a comment for what this magic number  `10` is and does, so that in the future we know why it's there and why it might need to be tweaked. Perhaps 10 won't be enough on a high enough resolution?